File: pattern-searching/ruby/naive.rb
class Naive
  def self.search(text, pattern)
    text_length = text.length
    pattern_length = pattern.length

    (0..text_length - pattern_length).each do |i|
      (0..pattern_length - 1).each do |j|
        break if text[i + j] != pattern[j]

        puts "Pattern found at position#{i}" if j == pattern_length - 1
      end
    end
  end
end

text = 'AABAACAADAABAAABAA'
pattern = 'AABA'

Naive.search(text, pattern)

File: pattern-searching/naive/dotnet/naive.cs
using System;

class Naive
{

    public static void search(String txt, String pat)
    {
        int M = pat.Length;
        int N = txt.Length;

        for (int i = 0; i <= N - M; i++)
        {
            int j;

            for (j = 0; j < M; j++)
                if (txt[i + j] != pat[j])
                    break;

            if (j == M)
                Console.WriteLine("Pattern found at index " + i);
        }
    }
}